cli/dstack/backend/base/__init__.py

```diff
@@ -240,6 +243,37 @@ def get_configurator(cls) -> Configurator:
     def delete_workflow_cache(self, workflow_name: str):
         pass
 
+    def run_workflow(
+        self,
+        workflow_name: str,
+        tag_name: Optional[str] = None,
+        ssh_pub_key: Optional[str] = None,
+        args: Optional[argparse.Namespace] = None,
+    ) -> (str, List[Job]):
+        """Runs workflow or provider by name
+        :return: run_name, jobs
+        """
+        workflow = self.repo.get_workflows(credentials=self.get_repo_credentials()).get(
+            workflow_name, {}
+        )
+        if workflow:
+            provider = providers.load_provider(workflow["provider"])
+        elif workflow_name in providers.get_provider_names():
+            provider = providers.load_provider(workflow_name)
+            workflow_name = None
+        else:
+            raise NameError(f"No workflow or provider '{workflow_name}' is found")
+
+        run_name = self.create_run()
+        provider.load(self, args, workflow_name, workflow, run_name, ssh_pub_key)
```

Comment:
Seems like it's not that hard to move deps resolution logic to backend and avoid passing backend to provider.